aip/client-libraries/4221.md

```diff
@@ -0,0 +1,80 @@
+---
+aip:
+  id: 4221
+  scope: client-libraries
+  state: draft
+  created: 2019-06-17
+permalink: /client-libraries/4221
+redirect_from:
+  - /4221
+---
+
+# Client-side retry
+
+Many APIs have error modes that a client should retry. These error modes vary
+across APIs, leaving users to hand-write retry logic around client libraries
+based on available documentation.
+
+## Guidance
+
+Client libraries **may** provide automatic client-side retry logic. Client
+libraries with automatic client-side retry logic **should** provide mechanisms
+(e.g. client initialization options) for users to configure it.
+
+Client libraries for API systems that support remotely-resolved client retry 
+configuration **should** respect the remotely-resolved configuration. However,
+user configuration via the aforementioned mechanisms **must** be respected.
+
+Client library generators implementing this feature **must** accept a retry
```

Comment:
"a retry configuration"?

Reply:
We probably need an example here.

Reply:
Done.

Reply:
I think we still probably need an example of what this looks like.

Reply:
Is the example in the specific section not sufficient? Are we looking for an example in this section too?

Reply:
Whups, sorry. I forgot to delete that comment when I came across the example.